api/src/main/java/io/strimzi/api/kafka/model/status/Credential.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.status;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.strimzi.api.kafka.model.UnknownPropertyPreserving;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+
+/**
+ * Represents a single listener
+ */
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({ "credentialsSecret", "credentials" })
+@EqualsAndHashCode
+public class Credential implements UnknownPropertyPreserving, Serializable {
```

Comment:
Should we add a field with the username? The username is not always just the name of the KafkaUser resource but it will be something like `CN=<resourceName>` for TLS. So it might make sense to list it here?

Reply:
I agree that's a good idea, but wouldn't that more naturally be a property of `KafkaUserStatus`?

Reply:
Yeah, I guess I just left it in wrong place.

Reply:
Why not here in `Credential`? To be honest I thought it was more natural to have it here. Now, this class has just the `Secret` name, is really useful to have it? If `userName` isn't here, maybe we can have a `userSecret` at the same level in the conditions instead of a `credential` field.